chain/syncer.go

```diff
@@ -195,12 +205,9 @@ func (syncer *Syncer) collectChain(ctx context.Context, tipsetCids types.TipSetK
 			return nil, ErrChainHasBadTipSet
 		}
 
-		blks, err := syncer.getBlksMaybeFromNet(ctx, tipsetCids.ToSlice())
-		if err != nil {
-			return nil, err
-		}
-
-		ts, err := types.NewTipSet(blks...)
+		// This will return a single tipset, error if not found and panic
+		// if more than 1 tipset is returned
+		ts, err := syncer.getBlksMaybeFromNet(ctx, tipsetCids.ToSlice())
```

Comment:
nit: I got a bit confused going off of the comment and var name.  I think it should say "This will return a slice containing a single tipset" and `ts` should probably be `tss` or some name indicating multiple values.

Reply:
I'll just make it return a single tipset like it did before.